Python/wolfssl-python-3.8.5.patch

```diff
@@ -0,0 +1,1995 @@
+ 
+         /* check to see if we've gotten to a new RDN */
+         if (rdn_level >= 0) {
++            #ifndef HAVE_WOLFSSL
+             if (rdn_level != X509_NAME_ENTRY_set(entry)) {
++            #endif
+                 /* yes, new RDN */
+                 /* add old RDN to DN */
+                 rdnt = PyList_AsTuple(rdn);
+@@ -1237,7 +1355,9 @@
+                 rdn = PyList_New(0);
+                 if (rdn == NULL)
+                     goto fail0;
++            #ifndef HAVE_WOLFSSL
+             }
++            #endif
+         }
+         rdn_level = X509_NAME_ENTRY_set(entry);
+ 
+@@ -1441,14 +1562,14 @@
+                 PyTuple_SET_ITEM(t, 0, v);
+ 
+                 if (name->d.ip->length == 4) {
+-                    unsigned char *p = name->d.ip->data;
++                    unsigned char *p = (unsigned char*)name->d.ip->data;
+                     v = PyUnicode_FromFormat(
+                         "%d.%d.%d.%d",
+                         p[0], p[1], p[2], p[3]
+                     );
+                 } else if (name->d.ip->length == 16) {
+                     /* PyUnicode_FromFormat() does not support %X */
+-                    unsigned char *p = name->d.ip->data;
++                    unsigned char *p = (unsigned char*)name->d.ip->data;
+                     len = sprintf(
+                         buf,
+                         "%X:%X:%X:%X:%X:%X:%X:%X",
+@@ -1559,8 +1680,21 @@
+     PyObject *lst = NULL, *ostr = NULL;
+     int i, result;
+     AUTHORITY_INFO_ACCESS *info;
+-
++    #ifdef HAVE_WOLFSSL
++    X509_EXTENSION* ext;
++    int loc = 0;
++    #endif
++    
++    #ifdef HAVE_WOLFSSL
++    loc = X509_get_ext_by_NID(certificate, NID_info_access, -1);
++    if (loc < 0)
++        return Py_None;
++    if((ext = X509_get_ext(certificate, loc)) == NULL)
++        return Py_None;
++    info = X509V3_EXT_d2i(ext);
++    #else
+     info = X509_get_ext_d2i(certificate, NID_info_access, NULL, NULL);
++    #endif
+     if (info == NULL)
+         return Py_None;
+     if (sk_ACCESS_DESCRIPTION_num(info) == 0) {
+@@ -1792,6 +1926,7 @@
+         Py_DECREF(peer_alt_names);
+     }
+ 
++
+     /* Authority Information Access: OCSP URIs */
+     obj = _get_aia_uri(certificate, NID_ad_OCSP);
+     if (obj == NULL) {
+@@ -1931,8 +2066,9 @@
+         return NULL;
+     }
+     peer_cert = SSL_get_peer_certificate(self->ssl);
+-    if (peer_cert == NULL)
++    if (peer_cert == NULL) {
+         Py_RETURN_NONE;
++    }
+ 
+     if (binary_mode) {
+         /* return cert in DER-encoded format */
+@@ -2154,7 +2290,11 @@
+ 
+     SSL_get0_alpn_selected(self->ssl, &out, &outlen);
+ 
++#ifdef HAVE_WOLFSSL
++    if (out == NULL || outlen == 0)
++#else
+     if (out == NULL)
++#endif
+         Py_RETURN_NONE;
+     return PyUnicode_FromStringAndSize((char *)out, outlen);
+ }
+@@ -2589,12 +2729,12 @@
+         PySSL_END_ALLOW_THREADS
+         self->err = err;
+ 
+-        if (PyErr_CheckSignals())
++        if (PyErr_CheckSignals()) {
```

Comment:
None of the changes in this hunk are needed. They don't hurt, but let's keep the patch as concise as we can, with only the required changes.